app/questionnaire/questionnaire_schema.py

```diff
@@ -734,6 +726,18 @@ def is_answer_in_repeating_section(self, answer_id: str) -> bool | None:
     def is_answer_dynamic(self, answer_id: str) -> bool:
         return answer_id in self._dynamic_answer_ids
 
+    def get_list_name_for_dynamic_answer(self, block_id: str) -> str | None:
+        return self.get_block(block_id).get("question", {}).get("dynamic_answers", {}).get("values", {}).get("identifier", {})  # type: ignore
```

Comment:
With the above this can be simplified as we will only be checking dynamic blocks

Reply:
Also just a minor point I would move the type ignore message above the line of code